app/Rules/Uppercase.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Rules;
+
+use Illuminate\Contracts\Validation\Rule;
+
+class Uppercase implements Rule
+{
+    /**
+     * Create a new rule instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+        $firstChar = explode('', $value)[0];
```

Comment:
The reason your tests are failing is here. You cannot pass in an empty delimiter to `explode`. In PHP 8 and above that will throw a value error. 7.4 and below it will trigger a warning. As shown in this 3v4l [demo](https://3v4l.org/CdJoF).

You can grab the first letter using the square bracket syntax just like arrays. 

```suggestion
        $firstChar = $value[0];
```

Reply:
@thinkverse you are hero :) 

i tried it and it worked. 

Lesson learnt....

Reply:
Glad to help.